sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/LogicalRelation.scala

```diff
@@ -72,9 +72,12 @@ case class LogicalRelation(
   // expId can be different but the relation is still the same.
   override lazy val cleanArgs: Seq[Any] = Seq(relation)
 
-  @transient override lazy val statistics: Statistics = Statistics(
-    sizeInBytes = BigInt(relation.sizeInBytes)
-  )
+  // statistics inherited from a MetastoreRelation
+  @transient var inheritedStats: Option[Statistics] = None
```

Comment:
Can you explain more about this comment? By "block it", do you mean don't compute stats for In-memory cataloged tables? Do they correspond to InMemoryRelation?